eva/executor/load_csv_executor.py

```diff
@@ -0,0 +1,76 @@
+# coding=utf-8
+# Copyright 2018-2020 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import numpy as np
+import pandas as pd
+
+from eva.configuration.configuration_manager import ConfigurationManager
+from eva.executor.abstract_executor import AbstractExecutor
+from eva.models.storage.batch import Batch
+from eva.planner.load_data_plan import LoadDataPlan
+from eva.storage.storage_engine import StorageEngine
+from eva.readers.csv_reader import CSVReader
+
+
+class LoadCSVExecutor(AbstractExecutor):
+
+    def __init__(self, node: LoadDataPlan):
+        super().__init__(node)
+        config = ConfigurationManager()
+        self.path_prefix = config.get_value('storage', 'path_prefix')
+
+    def validate(self):
+        pass
+
+    def exec(self):
+        """
+        Read the input meta file using pandas and persist data
+        using storage engine
+        """
+
+        # TODO: Should we move this to utils?
+        # utility function to convert bbox to lisst of floats
+        def convert_bbox(bbox):
+            return np.array([np.float32(coord) for coord in bbox.split(",")])
+
+        # Read the CSV file
+        # converters is a dictionary of functions that convert the values
+        # in the column to the desired type
+        # column_list is a list of columns to be read from the CSV file. 
+        # This should match the table schema
+        csv_reader = CSVReader(
+            os.path.join(self.path_prefix, self.node.file_path),
+            converters={"bbox": convert_bbox},
```

Comment:
It's better if converters can be configurable instead of hard coded.

Reply:
I gave this some thought too. When you say 'configurable', you want it to be configured through the query itself? Can you give an example? 

The reason why I had to add this converter was because, by default, when pandas reads a CSV file, it inputs columns values as string values.

Reply:
Yeah, the idea is that we can have converter as something like UDF, and in query when we specify the columns list, we can also specify the converter if needed.  

Like if the format you pick for columns is LOAD DATA INFILE 'test.csv' INTO meta_bdd WITH FORMAT CSV COLUMNS ('id', 'frame_id', 'video_id', 'dataset_name', 'label', 'bbox', 'object_id'). Then we can have LOAD DATA INFILE 'test.csv' INTO meta_bdd WITH FORMAT CSV COLUMNS ('id', 'frame_id', 'video_id', 'dataset_name', 'label', 'bbox': 'bbox_converter', 'object_id')

Reply:
Oh so bbox_converter should be a registered UDF then?

Reply:
uhm. This is good point. UDF actually inputs dataframe and outputs dataframe. I believe converter inputs string and outputs numpy array. Now i think about it, the better way to do that is have this convert as a built-in function like what you did. And have a syntax sugar on the query to indicate which column needs to be converted (so it doesn't work only on the bbox column.). For example  LOAD DATA INFILE 'test.csv' INTO meta_bdd WITH FORMAT CSV COLUMNS ('id', 'frame_id', 'video_id', 'dataset_name', 'label', NDARRAY('bbox'), 'object_id'). @gaurav274 What is your opinion?